src/utils/ucx/ucx_utils.cpp

```diff
@@ -406,15 +406,15 @@ nixlUcxWorker::nixlUcxWorker(std::shared_ptr<nixlUcxContext> &_ctx): ctx(_ctx)
         worker_params.thread_mode = UCS_THREAD_MODE_MULTI;
         break;
     default:
-        assert(ctx->mt_type < NIXL_UCX_MT_MAX);
-        abort();
+        NIXL_FATAL << "Invalid UCX worker type: " << ctx->mt_type;
     }
 
     status = ucp_worker_create(ctx->ctx, &worker_params, &worker);
-    if (status != UCS_OK)
-    {
-       // TODO: MSW_NET_ERROR(priv->net, "failed to create ucp_worker (%s)\n", ucs_status_string(status));
-        return;
+    if (status != UCS_OK) {
+        auto err_str = std::string("Failed to create UCX worker: ") +
+                       ucs_status_string(status);
+        NIXL_ERROR << err_str;
+        throw std::runtime_error(err_str);
```

Comment:
maybe ` throw std::exception();` and no need for `err_str`, as you do not use it anyway

Reply:
I used the same exception type as existing one for `ucp_ep_create`

Reply:
Also `std::runtime_error` does not have default ctor

Reply:
seems odd to generate an error string which you are not going to use. Can't you use default `std::exception`?

Reply:
I can, but IMO better to have it aligned